### PR TITLE
Add administration for banners

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Admin/HomeBlockAdmin.php [hunk 1]
```diff
@@ -89,6 +90,24 @@ protected function configureFormFields(FormMapper $formMapper)
...
             ;
         }
+
+        if (HomeBlock::TYPE_BANNER === $this->getSubject()->getType()) {
+            $formMapper
+                ->add('titleCta', null, [
+                    'label' => 'Texte du CTA',
+                    'required' => false,
+                ])
+                ->add('colorCta', ChoiceType::class, [
```
Comment (tgalopin) [inline src/Admin/HomeBlockAdmin.php]: Could you use an input text to put the hexadecimal code instead?

File: src/Admin/Color.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace AppBundle\Admin;
+
+class Color
+{
+    const COLOR_BLACK = 'black';
```
Comment (HeahDude) [inline src/Admin/Color.php]: It would be better to have `Color::BLACK` than `Color::COLOR_BLACK`.

File: src/Admin/HomeBlockAdmin.php [hunk 3]
```diff
@@ -89,6 +90,24 @@ protected function configureFormFields(FormMapper $formMapper)
                 ])
             ;
         }
+
+        if (HomeBlock::TYPE_BANNER === $this->getSubject()->getType()) {
+            $formMapper
+                ->add('titleCta', null, [
```
Comment (HeahDude) [inline src/Admin/HomeBlockAdmin.php]: We prefer snake case for field names by convention.

File: src/Entity/HomeBlock.php [hunk 4]
```diff
@@ -131,15 +154,15 @@ public function getId()
     }
 
     /**
-     * @return null|string
+     * @return int
```
Comment (HeahDude) [inline src/Entity/HomeBlock.php]: Type hints should be used instead of phpdoc when possible.

File: src/Entity/HomeBlock.php [hunk 5]
```diff
@@ -282,6 +305,36 @@ public function setDisplayTitles(bool $displayTitles)
         $this->displayTitles = $displayTitles;
     }
 
+    public function isDisplayBlock(): bool
+    {
+        return $this->displayBlock;
+    }
+
+    public function setDisplayBlock(bool $displayBlock)
```
Comment (HeahDude) [inline src/Entity/HomeBlock.php]: `: void`

File: src/Entity/HomeBlock.php [hunk 6]
```diff
@@ -282,6 +305,36 @@ public function setDisplayTitles(bool $displayTitles)
...
+    {
+        $this->displayBlock = $displayBlock;
+    }
+
+    public function getTitleCta()
```
Comment (HeahDude) [inline src/Entity/HomeBlock.php]: `: ?string`

File: src/Entity/HomeBlock.php [hunk 7]
```diff
@@ -282,6 +305,36 @@ public function setDisplayTitles(bool $displayTitles)
...
+    {
+        return $this->titleCta;
+    }
+
+    public function setTitleCta($titleCta)
```
Comment (HeahDude) [inline src/Entity/HomeBlock.php]: `public function setTitleCta(?string $titleCta): void`

and same below.

File: src/Admin/HomeBlockAdmin.php (hunk 1 above)
Comment (illusionOfParadise) [inline src/Admin/HomeBlockAdmin.php]: @tgalopin I have a list of colors (specified in the ticket), and more it matches the class names. Do you really prefer to use text input ?

Comment (tgalopin) [inline src/Admin/HomeBlockAdmin.php]: Oh OK, let's go that way then.

**Pull request merged**